### PR TITLE
Fix handling of numeric weekdays (Fixes #74)

### LINKED ISSUE #74: Support Weekdays
In cron tab you can do something like 0-2 or mon-fri . It seems the validation doesn't allow for this. The field should follow:

```
 # ┌───────────── min (0 - 59) 
 # │ ┌────────────── hour (0 - 23)
 # │ │ ┌─────────────── day of month (1 - 31)
 # │ │ │ ┌──────────────── month (1 - 12)
 # │ │ │ │ ┌───────────────── day of week (0 - 6) (0 to 6 are Sunday to Saturday, or use names; 7 is Sunday, the same as 0)
 # │ │ │ │ │
 # │ │ │ │ │
 # * * * * *  command to execute
```

### PR DESCRIPTION
**Pull request by eherot**
### Description

Match integers using duck typing instead of `is_a? Integer` which doesn't actually work.

### Issues Resolved

- #74: Support Weekdays
- #90: cron_d month option not accepting integer. (basically a duplicate of #74)

### Check List

- [✓] All tests pass.* See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [✓] New functionality includes testing.
- [NA] New functionality has been documented in the README if applicable
- [✓] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

*Tests pass once #92 is merged in ;-)

### CODE REVIEW

**Comment by eherot** [comment]
This is going to have a merge conflict with #92. Ping me once merged and I'll fix it.

**Comment by Stromweld** [comment]
@eherot you need to add "Signed-off-by: Julia Child <juliachild@chef.io>" to the body or end of line in your commit message to get it to be DCO signed-off. Basically saying you lay no claims to code committed.

**Comment by eherot** [comment]
Yep. Fixing that now.

**Comment by Stromweld** [comment]
Looking at the code that got changed when it was switched from a LWRP to a resource, the line for validate_nemeric method and validate_month that changed and created the breakage is "if spec.is_a? Fixnum" to "if spec.is_a? Integer".

Either this PR should fix it or could simply switch it back from Integer to Fixnum. @tas50

**Comment by eherot** [comment]
I opted for duck typing vs. exact type matching so that weird types like `Chef::Node::ImmutableMash` wouldn't trip it up if they happened to find their way in there.

**Comment by eherot** [comment]
Note that the TravisCI tests are not going to pass until #92 is merged (in case that wasn't obvious).

**Comment by tas50** [comment]
@eherot Can you rebase this one so I can merge it in?

**Comment by eherot** [comment]
coming right up!

File: libraries/helpers.rb [hunk 1]
```diff
@@ -50,9 +50,9 @@ def self.validate_month(spec)
 
     def self.validate_dow(spec)
       return true if spec == '*'
-      if spec.class == Integer
+      if spec.respond_to? :to_int
         validate_numeric(spec, 0, 7)
-      elsif spec.class == String
+      elsif spec.respond_to? :to_str
```
Comment (eherot) [inline libraries/helpers.rb]: @tas50 Also changed this for consistency's sake.

**Comment by eherot** [comment]
Rebased. Also made a small change I sent you a note about I just want to make sure you're ok with.

**Comment by eherot** [comment]
Thanks! If you could cut a release soon-ish that would save me having to write a workaround :-)

**Pull request merged**